src/Backups/BackupEntriesCollector.cpp

```diff
@@ -781,6 +781,17 @@ void BackupEntriesCollector::makeBackupEntriesForTablesDefs()
 
         const String & metadata_path_in_backup = table_info.metadata_path_in_backup;
         backup_entries.emplace_back(metadata_path_in_backup, std::make_shared<BackupEntryFromMemory>(new_create_query->formatWithSecretsOneLine()));
+
+        // metadata_version only makes sense for replicated tables.
+        // Process metadata_version separately from the parts
+        // to ensure it's captured regardless of part processing (e.g. structure_only=1)
+        if (table_info.storage->getName().starts_with("Replicated"))
```

Comment:
may be 
table_info.storage->supportsReplication()
?

Reply:
Well ... probably using name prefix is less errorprone. At least because Engine 'Proxy' exists.

Reply:
Well, maybe. But I am not sure if it would be any better.

I see that StorageDummy and StorageProxy `supportsReplication()` can be true. Aren't there any negative effects? E.g. at some point we may attempt to back up storageDummy. Shall we take their metadata_version into account? It is only important and properly managed in ReplicatedMTs

Reply:
Yes, that is exactly why I 'revoke' my suggestion  about supportsReplication.
Using name prefix seems better approach.